### PR TITLE
lib: compress: Use libarchive

### PR DESCRIPTION
**Pull request by ernestask**
Currently, the library only handles LZMA- and LZ4-compressed archives by
using separate libraries for each. This commit adopts the use of
libarchive to handle a plethora of compression formats, including
Zstandard, which is available in systemd since v246-rc1.

https://bugzilla.redhat.com/show_bug.cgi?id=1860616

### CODE REVIEW

File: src/lib/compress.c [hunk 1]
```diff
@@ -18,190 +18,85 @@
...
-{
-    uint8_t buf_in[BUFSIZ];
-    uint8_t buf_out[BUFSIZ];
+    archive_read_support_filter_all(archive);
+    archive_read_support_format_raw(archive);
 
-    lzma_stream strm = LZMA_STREAM_INIT;
-    lzma_ret ret = lzma_stream_decoder(&strm, UINT64_MAX, 0);
-    if (ret != LZMA_OK)
+    r = archive_read_open_fd(archive, fdi, 16384);
```
Comment (mgrabovsky) [inline src/lib/compress.c:36]: Can you please factor the block/buffer size out into a named constant? It wasn't immediately apparent to me what the number designated.

**Comment by mgrabovsky** [comment]
I'm getting a segfault on the decompression test on `make rpm`. Same on a fresh Rawhide VM.
```
Program terminated with signal SIGABRT, Aborted.
#0  0x00007f88ea0e3b95 in raise () from /lib64/libc.so.6
(gdb) bt full
#0  0x00007f88ea0e3b95 in raise () at /lib64/libc.so.6
#1  0x00007f88ea0cc8a4 in abort () at /lib64/libc.so.6
#2  0x00007f88ea290b6c in g_assertion_message_expr.cold () at /lib64/libglib-2.0.so.0
#3  0x00007f88ea2ef3ea in g_assertion_message_cmpnum () at /lib64/libglib-2.0.so.0
#4  0x00005615709b67ff in test_decompression (user_data=<optimized out>) at compress.c:15
        __n1 = -1
        template = "/tmp/libreport-test_decompression-XXXXXX"
        in_fd = -1
        out_fd = <optimized out>
        result = <optimized out>
        __func__ = "test_decompression"
#5  0x00007f88ea2ee92e in g_test_run_suite_internal () at /lib64/libglib-2.0.so.0
#6  0x00007f88ea2ee72b in g_test_run_suite_internal () at /lib64/libglib-2.0.so.0
#7  0x00007f88ea2eee1a in g_test_run_suite () at /lib64/libglib-2.0.so.0
#8  0x00007f88ea2eee35 in g_test_run () at /lib64/libglib-2.0.so.0
#9  0x00005615709b6284 in main (argc=<optimized out>, argv=<optimized out>) at compress.c:67
```

**Comment by ernestask** [comment]
Try now.

**Comment by mgrabovsky** [comment]
Tests run fine now, but I get a bunch of the following errors for `dnf upgrade`:
```
nothing provides libarchive{?_isa} needed by libreport-2.13.1.92.ge234.dirty-1.fc33.x86_64
```

**Comment by ernestask** [comment]
How I hate RPM. Should be good now.

**Pull request merged**